code/datums/viewport.dm

```diff
@@ -0,0 +1,100 @@
+#define ceil(x) (-round(-(x)))
+/client
+	var/obj/skybox/skybox
+	var/last_view_x_dim = 7
+	var/last_view_y_dim = 7
+
+/client/verb/SetWindowIconSize(var/val as num|text)
+	set hidden = 1
+	winset(src, "mapwindow.map", "icon-size=[val]")
+	if(val != prefs?.icon_size)
+		prefs.icon_size = val
+		prefs.save_preferences()
+	OnResize()
+
+/client/verb/OnResize()
+	set hidden = 1
+	if(!prefs.read_preference(/datum/preference/toggle/widescreen))
+		return
+
+	var/divisor = text2num(winget(src, "mapwindow.map", "icon-size")) || world.icon_size
+	if(!isnull(config.lock_client_view_x) && !isnull(config.lock_client_view_y))
+		last_view_x_dim = config.lock_client_view_x
+		last_view_y_dim = config.lock_client_view_y
+	else
+		var/winsize_string = winget(src, "mapwindow.map", "size")
+		last_view_x_dim = config.lock_client_view_x || clamp(ceil(text2num(winsize_string) / divisor), 15, (CONFIG_GET(number/max_client_view_x)) || 41)
+		last_view_y_dim = config.lock_client_view_y || clamp(ceil(text2num(copytext(winsize_string,findtext(winsize_string,"x")+1,0)) / divisor), 15, (CONFIG_GET(number/max_client_view_y)) || 41)
+		if(last_view_x_dim % 2 == 0) last_view_x_dim++
+		if(last_view_y_dim % 2 == 0) last_view_y_dim++
+	for(var/check_icon_size in global.valid_icon_sizes)
+		winset(src, "menu.icon[check_icon_size]", "is-checked=false")
+	winset(src, "menu.icon[divisor]", "is-checked=true")
+
+	view = "[last_view_x_dim]x[last_view_y_dim]"
+
+	// Reset eye/perspective
+	var/last_perspective = perspective
+	perspective = MOB_PERSPECTIVE
+	if(perspective != last_perspective)
+		perspective = last_perspective
+	var/last_eye = eye
+	eye = mob
+	if(eye != last_eye)
+		eye = last_eye
+
+	// Recenter skybox and lighting.
+	mob?.reload_fullscreen()
+	// We do this aswell because things tend to be buggy if not. Mostly lighting and sprite origin issues.
+	change_view(getScreenSize(prefs.read_preference(/datum/preference/toggle/widescreen)))
+
+/client/verb/force_onresize_view_update()
```

Comment:
These verbs don't seem to be applied anyway, are we adding new debug verbs to every player? I don't think I want that.